anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,407 @@
+import json
+
+from anchore_engine.utils import ensure_str
+
+
+class HintsError(Exception):
+    pass
+
+
+class Hint:
+    def __init__(self, pkg, type):
+        self.type = type
+        self.name = ensure_str(pkg.get("name", ""))
+        self.version = ensure_str(pkg.get("version", ""))
+        self.origin = ensure_str(pkg.get("origin", ""))
+
+    def check_required_fields(self):
+        if not self.name or not self.version or not self.type:
+            raise HintsError(
+                "bad hints record, all hints records must supply at least a name, version and type"
+            )
+
+    def normalize(self):
+        pass
+
+    @staticmethod
+    def get_list_value(pkg, key):
+        value = ensure_str(pkg.get(key, ""))
+
+        # For Gem/NPM hints, we search both the singular and plural key name, where the plural is a list
+        list_value = pkg.get(key + "s", [])
```

Comment:
this looks potentially brittle, what if `key` is a `None`? or an empty string?

Reply:
added check in beginning of method and case to unit test